.github/workflows/publish-on-pypi.yml

```diff
@@ -5,6 +5,13 @@ on:
     tags:
       # After vMajor.Minor.Patch _anything_ is allowed (without "/") !
       - v[0-9]+.[0-9]+.[0-9]+*
+    branches:
```

Comment:
I don't know that this will work, but the `release` should do it, I belieeve.

Reply:
The combination of both do it.

Reply:
Could we test it with some a pre-release tag?

Reply:
(after merging)

Reply:
We can just do a patch release `v0.7.1`

Reply:
> The combination of both do it.

I'm not sure the github actions logic work this way, is my only concern. It's not AND statements, but rather OR as I understand it, but I may definitely be wrong, since it also seems to update a bit at the moment.

Reply:
Seeing the actions run ([here](https://github.com/Materials-Consortia/optimade-python-tools/runs/514042313)) it seems clear it's an OR logic, i.e., the workflow ran on a push to the branch `master`. But didn't release on PyPI only due to the check in the code that the version in `__init__.py` doesn't match.